### PR TITLE
Proposed solution

### PR DESCRIPTION
**Pull request by ImMPrada**
This PR adds a proposed solution and a README.

You can read [README here](https://github.com/ImMPrada/top_bubble_sort/blob/7e82b29967ed20e5b1da466ea6d0cd8cb115e6d9/README.md), to get involved about the challenge

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<img width="226" alt="image" src="https://user-images.githubusercontent.com/26731448/188519846-a027234f-8008-40cb-8b22-87a233650de0.png">
+
+
+## Bubble sort
+
+A challenge on [Odin's project](https://www.theodinproject.com/lessons/ruby-bubble-sort)
```
Comment (mapra99) [inline README.md:6]:
```suggestion
A challenge on [Odin Project](https://www.theodinproject.com/lessons/ruby-bubble-sort)
```

File: lib/main.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+def bubble_sort(numbers)
+  sorted = true
+
+  (numbers.length - 1).times do |i|
```
Comment (mapra99) [inline lib/main.rb]: This is just a personal preference. Leave it or take it :P
```suggestion
  (0...numbers.length).each do |i|
```

File: lib/main.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    sub_arr = [numbers[i], numbers[i + 1]]
+
+    next if sub_arr[0] <= sub_arr[1]
+
+    numbers = numbers[0, i] + sub_arr.sort! + numbers[i + 2...]
```
Comment (mapra99) [inline lib/main.rb]: using `sort!` in a sorting algorithm must be cheating 🤣. Why not something like this?
```suggestion
    numbers[i] = sub_arr[1]
    numbers[i + 1] = sub_arr[0]
```

File: lib/main.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    sorted = false
+    break
```
Comment (mapra99) [inline lib/main.rb]: I think this `break` shouldn't be here. See on the gif in the readme, that the algorithm compares each pair of numbers up from the beginning until the end of the array, and then it starts again on the beginning. Here, this `break` statement is not letting the comparison get to the end of the array; instead, it gets only to the first pair that requires a position swap, then the iteration breaks and the recursive call makes the iteration start over from the beginning of the array

File: lib/main.rb (hunk 3 above)
Comment (ImMPrada) [reply lib/main.rb]: jajajajjajajajjajajaja yeah, better! ajjajjajajajajjaajjajjaja

File: lib/main.rb (hunk 2 above)
Comment (ImMPrada) [reply lib/main.rb]: take a look!!

File: lib/main.rb (hunk 4 above)
Comment (ImMPrada) [reply lib/main.rb]: I agree. With that "break", I'll be restarting the break each we sort a pair of numbers. But I need to go through all the array's indexes, sorting each pair of numbers

image

**Review by mapra99 — APPROVED** [review]

**Pull request merged**